### DISCUSSION TITLE
empty backup file

### DISCUSSION THREAD

**Discussion by SanChriTa** [teslamate-org/teslamate#3400] [General]
I am a beginner I don't grasp Docker very well. I have version 1.27.1 from the beginning and Tesla disappeared after updating its software. I wanted to update Tesla Mate to the latest version 1.27.3, and back up before that. After logging into the server via ssh, I used this: `docker compose exec -T database pg_dump -U teslamate teslamate > /backuplocation/teslamate.bck`

I got these messages and after downloading the bck file it is empty.
WARN[0000] The "FQDN_TM" variable is not set. Defaulting to a blank string.
WARN[0000] The "FQDN_TM" variable is not set. Defaulting to a blank string.
WARN[0000] The "LETSENCRYPT_EMAIL" variable is not set. Defaulting to a blank string.
http: invalid host header

Where to find the configuration with these variables and could this be the reason that the backup has a size of 0B?
And will it be enough later 
```
docker compose pull
docker compose up -d
``` 

to update the version?

**Comment by JakobLichterfeld** [comment]
Hi,
please follow the steps in: https://docs.teslamate.org/docs/maintenance/backup_restore#backup and use your values you set in your docker-compose.yml as stated in the notes.

**Reply by reuten75** [reply]
Hi! I'm trying to get this but I am a really newbie in databases and etc. The standard command does not work for me. I have the standard user teslamate and nothing is changed except that my compose file is situated under a portainer folder in my docker installation. Don't if that affects anything.
This is the output from the power shell: "unknown shorthand flag: 'T' in -T
So that is not working for me.

Help is appreciated.

BG
Henrik

**Reply by JakobLichterfeld** [reply]
> under a portainer folder in my docker installation. Don't if that affects anything.

Which system are you using? Seems like you use Portainer in addition, which affects everything.
OU can omit the -T in that case I assume, but we do not give Portainer support.
